src/main/java/seedu/mark/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,153 @@
+    public static final String MESSAGE_FILE_FORMAT_INCORRECT = "The format of data in the file %1$s is incorrect";
+    public static final String MESSAGE_FILE_NOT_FOUND = "The file %1$s does not exist";
+
+    private static final Logger logger = LogsCenter.getLogger(JsonMarkStorage.class);
+
+    private final Path filePath;
+
+    /**
+     * Creates a new {@code ImportCommand} instance with the given file path.
+     *
+     * @param filePath of the file to be imported from.
+     */
+    public ImportCommand(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Converts a given list of {@code Bookmark}s to a multi-line {@code String},
+     * with each line having an indent of four spaces.
+     *
+     * @param bookmarks List of bookmarks.
+     * @return String representation of the bookmark list with indentation.
+     */
+    public static String toIndentedString(List<Bookmark> bookmarks) {
+        String newlineAndIndent = "\n    ";
+        return bookmarks.stream().map(Bookmark::toString)
+                .map(bookmark -> newlineAndIndent.concat(bookmark))
+                .reduce("", String::concat);
+    }
+
+    /**
+     * Executes the import command on the given {@code model} and {@code storage}.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @param storage {@code Storage} which the command should operate on.
+     * @return A CommandResult indicating a successful import, and (if applicable)
+     *         shows bookmarks that were skipped during import.
+     * @throws CommandException if an error occurs while reading bookmarks from Storage
+     */
+    @Override
+    public CommandResult execute(Model model, Storage storage) throws CommandException {
+        requireAllNonNull(model, storage);
+
+        ReadOnlyMark newMark = readMarkFromStorageFile(storage, filePath);
+
+        ObservableList<Bookmark> bookmarksToImport = newMark.getBookmarkList();
+        List<Bookmark> skippedBookmarks = new ArrayList<>(); // empty list to operate on
+        addBookmarksToModel(model, bookmarksToImport, skippedBookmarks);
+
+        if (!skippedBookmarks.isEmpty()) {
+            logger.info("Bookmarks imported from " + filePath + ": Some duplicates skipped");
+            return new CommandResult(String.format(MESSAGE_IMPORT_SUCCESS_WITH_DUPLICATES, filePath,
+                    toIndentedString(skippedBookmarks)));
+        }
+
+        logger.info("Bookmarks imported from " + filePath);
+        return new CommandResult(String.format(MESSAGE_IMPORT_SUCCESS, filePath));
+    }
+
+    /**
+     * Attempts to read Mark from a given file in storage.
+     */
+    private ReadOnlyMark readMarkFromStorageFile(Storage storage, Path filePath) throws CommandException {
+        requireAllNonNull(storage, filePath);
+
+        Optional<ReadOnlyMark> newMark;
+        try {
+            newMark = storage.readMark(filePath);
+        } catch (IOException ioe) {
```

Comment:
I don't think IOException can be thrown either:

![image](https://user-images.githubusercontent.com/3646725/66658197-906d7f00-ec73-11e9-9b7c-8b9afd25864a.png)

Reply:
Hm the `Storage` interface allows for an `IOException` to be thrown though:

![image](https://user-images.githubusercontent.com/46101221/66700714-4d201880-ed26-11e9-8722-6a907d43b6cc.png)

Reply:
Ah yes, my mistake:

JsonMarkStorage, the one we're using, doesn't throw IOException.

but the interface Storage does allow IOException to be thrown.